flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/DateTimeUtils.java

```diff
@@ -1251,133 +1356,24 @@ private static long julianDateFloor(TimeUnitRange range, int julian, boolean flo
-     *
-     * @param dateStr formatted date string.
-     * @param days days count you want to subtract.
-     * @param tz time zone of the date time string
-     * @return datetime string.
-     */
-    public static String dateSub(String dateStr, int days, TimeZone tz) {
-        long ts = parseToTimeMillis(dateStr, tz);
-        if (ts == Long.MIN_VALUE) {
-            return null;
-        }
-        return dateSub(ts, days, tz);
-    }
-
-    /**
-     * Do subtraction on date string.
-     *
-     * @param ts the timestamp.
-     * @param days days count you want to subtract.
-     * @param tz time zone of the date time string
-     * @return datetime string.
-     */
-    public static String dateSub(long ts, int days, TimeZone tz) {
-        ZoneId zoneId = tz.toZoneId();
-        Instant instant = Instant.ofEpochMilli(ts);
-        ZonedDateTime zdt = ZonedDateTime.ofInstant(instant, zoneId);
-        long resultTs = zdt.minusDays(days).toInstant().toEpochMilli();
-        return dateFormat(resultTs, DATE_FORMAT_STRING, tz);
-    }
-
-    public static String dateSub(String dateStr, int days) {
-        return dateSub(dateStr, days, UTC_ZONE);
-    }
-
-    public static String dateSub(long ts, int days) {
-        return dateSub(ts, days, UTC_ZONE);
-    }
-
     /**
-     * Do addition on date string.
+     * Convert datetime string from a time zone to another time zone.
      *
-     * @param dateStr formatted date string.
-     * @param days days count you want to add.
-     * @return datetime string.
+     * @param dateStr the date time string
+     * @param tzFrom the original time zone
+     * @param tzTo the target time zone
      */
-    public static String dateAdd(String dateStr, int days, TimeZone tz) {
-        long ts = parseToTimeMillis(dateStr, tz);
-        if (ts == Long.MIN_VALUE) {
+    public static String convertTz(String dateStr, String tzFrom, String tzTo) {
+        try {
+            return dateFormatTz(parseTimestampTz(dateStr, tzFrom), tzTo);
+        } catch (ParseException e) {
             return null;
         }
-        return dateAdd(ts, days, tz);
-    }
-
-    /**
-     * Do addition on timestamp.
-     *
-     * @param ts the timestamp.
-     * @param days days count you want to add.
-     * @return datetime string.
-     */
-    public static String dateAdd(long ts, int days, TimeZone tz) {
-        ZoneId zoneId = tz.toZoneId();
-        Instant instant = Instant.ofEpochMilli(ts);
-        ZonedDateTime zdt = ZonedDateTime.ofInstant(instant, zoneId);
-        long resultTs = zdt.plusDays(days).toInstant().toEpochMilli();
-        return dateFormat(resultTs, DATE_FORMAT_STRING, tz);
-    }
-
-    public static String dateAdd(String dateStr, int days) {
-        return dateAdd(dateStr, days, UTC_ZONE);
     }
 
-    public static String dateAdd(long ts, int days) {
-        return dateAdd(ts, days, UTC_ZONE);
+    private static String dateFormatTz(long ts, String tzStr) {
```

Comment:
Wouldn't `timestampFormatTz` be more appropriate name? `date` for me is confusing.

Reply:
Renamed `formatTimestampTz`